### ISSUE TITLE
TypeError (TS2740) sliderComponent 

### ISSUE THREAD

**Issue by Hektar90** [instea/react-native-color-picker#41]
I just want to use the default slider from '@react-native-community/slider'

<img width="386" alt="Bildschirmfoto 2021-02-22 um 17 56 39" src="https://user-images.githubusercontent.com/15243732/108741791-5da9eb00-7537-11eb-9b4e-4b1b47d8cc2d.png">

adding this to the 'sliderComponent' prop results in an type error

![Bildschirmfoto 2021-02-22 um 17 48 45](https://user-images.githubusercontent.com/15243732/108741636-38b57800-7537-11eb-9a9b-ad8b4e5c3d5f.png)

Slider is an intersection type of React.Component

<img width="785" alt="Bildschirmfoto 2021-02-22 um 17 59 52" src="https://user-images.githubusercontent.com/15243732/108743449-13296e00-7539-11eb-8fee-05d1191aa356.png">

**Comment by Frakette** [comment]
I have an issue too using '@react-native-community/slider' as sliderComponent. I don't get a crash but i think there is indeed an issue with the typechecking of sliderComponent props.
Here's an extract of my code : 
<ColorPicker
    style={{ flex: 1 }}
    defaultColor={newColor}
    onColorChange={color => setNewColor(fromHsv(color))}
    // not working for now - seems to be an error in code of the package - see latest issue in package
    sliderComponent={(props: any) => (
      <SliderComponent
        value={props.value}
        onValueCange={() => props.onValueChange()}
      />
    )}
/>

My SliderComponent get the value prop.
However, the onValueChange prop doesn't seem to work because when I slide, none of the "v" / "s" value are changing.

**Comment by Hektar90** [comment]
@Frakette Do you use typescript in your project? 
This is just a type error issue which means it compiles and works. Only my IDE is complaining.

SliderComponent isn't customized, is it? So just add the component class like so:

`<ColorPicker ...sliderComponent={SliderComponent }`

this should work if i get you right

**Comment by Frakette** [comment]
@Hektar90 Yes I use typescript on my project. Actually my issue might not be fully related to yours but I also see a Typescript error when I try to add a sliderComponent.

I was struggling to pass props to my SliderComponent because I was passing it directly as a component and it was not working but as a function that returns a component it works although the slider is a bit glitchy.

However, as you noticed, I still get get the same Typescript error than you when passing SliderComponent.
I ended up using a //@ts-expect-error to avoid compilation errors. But a correction on the type checking would be great :)

**Comment by gprinc-hu** [comment]
Any update in this issue? I have the same problem
